pennylane/qchem/openfermion_obs.py

```diff
@@ -945,11 +962,16 @@ def molecular_hamiltonian(
         if args is None:
             h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)()
             coeffs = qml.numpy.real(h.coeffs, requires_grad=False)
-            h = qml.Hamiltonian(coeffs, h.ops, grouping_type=grouping_type, method=grouping_method)
         else:
             h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)(*args)
             coeffs = qml.numpy.real(h.coeffs)
+
+        if active_new_opmath():
+            ps = qml.dot(coeffs, h.ops, pauli=True)
+            h = qml.s_prod(0, qml.Identity(h.ops[0].wires[0])) if len(ps) == 0 else ps.operation()
```

Comment:
I think we should modify all the tests to make them compatible with the `Sum` type.

Reply:
Adding now 👍🏼